tardis/energy_input/gamma_ray_transport.py

```diff
@@ -260,6 +260,85 @@ def initialize_packets(
     )
 
 
+def calculate_shell_masses(model):
+    """Function to calculate shell masses
+    Parameters
+    ----------
+    model : tardis.Radial1DModel
+        The tardis model to calculate gamma ray propagation through
+    Returns
+    -------
+    numpy.ndarray
+        shell masses in units of g
+
+    """
+    outer_velocities = model.v_outer.to("cm/s").value
+    inner_velocities = model.v_inner.to("cm/s").value
+    ejecta_density = model.density.to("g/cm^3").value
+    ejecta_volume = model.volume.to("cm^3").value
+    shell_masses = ejecta_volume * ejecta_density
+
+    return shell_masses
+
+
+def calculate_isotope_mass(model):
+    """Function to calculate isotope mass
+    Parameters
+    ----------
+    model : tardis.Radial1DModel
+        The tardis model to calculate gamma ray propagation through
+    Returns
+    -------
+        pandas dataframe of raw isotope masses in units of g
+    """
+    raw_isotope_abundance = model.raw_isotope_abundance.sort_values(
+        by=["atomic_number", "mass_number"], ascending=True
+    )
+
+    raw_isotope_mass = raw_isotope_abundance.apply(
+        lambda x: x * calculate_shell_masses(model), axis=1
+    )  # calculate masses of isotope
+
+    return raw_isotope_mass
+
+
+def create_inventories(model, time_end=80.0):
+
+    """Function to create inventories of isotope
+    Parameters
+    ----------
+    model : tardis.Radial1DModel
+        The tardis model to calculate gamma ray propagation through
+
+    time_end : float
+        End time of simulation in days
+    Returns
+    -------
+       None
+    """
+
+    time_end *= u.d.to(u.s)
+    raw_isotope_mass = calculate_isotope_mass(model)
+    inventories = (
+        raw_isotope_mass.to_inventories()
+    )  # inventories are in disintegrations for x g of isotope
+    shell_masses = calculate_shell_masses(model)
+
+    inventory_list = []
+    for shell, inv in enumerate(inventories):
+
+        decay_per_gram = (
+            inv / shell_masses[shell]
```

Comment:
`inv` is not enough to get the total decays. You need to use the method `cumulative_decays`. They are also already in per gram so you do not need to divide by shell masses again

Reply:
You can't divide an inventory by mass